server/container/guice/cassandra/src/main/java/org/apache/james/modules/mailbox/CassandraMailboxModule.java

```diff
@@ -226,7 +226,7 @@ protected void configure() {
         cassandraDataDefinitions.addBinding().toInstance(CassandraModSeqModule.MODULE);
         cassandraDataDefinitions.addBinding().toInstance(CassandraSubscriptionModule.MODULE);
         cassandraDataDefinitions.addBinding().toInstance(CassandraUidModule.MODULE);
-        cassandraDataDefinitions.addBinding().toInstance(CassandraThreadModule.MODULE);
+        cassandraDataDefinitions.addBinding().toInstance(CassandraThreadIdGuessingModule.MODULE);
```

Comment:
I meanm can we have a dedicated Guice module <3 ?

Reply:
do you mean separate it like this?
![image](https://user-images.githubusercontent.com/55171818/126770339-6c28796d-59bc-45dd-bddb-ce2978fb73fd.png)

Reply:
Or I should put ThreadId module here?
![image](https://user-images.githubusercontent.com/55171818/126771404-6c9c3a26-8fdc-47fd-b166-bc82d65c180b.png)

Reply:
Yes. To be honnest I think in some configuration supporting threads does not make sense.

(IMAP only)

Having it separated makes it easier to switch it off if needed.